File: platform-integration/ios/SecureCommunication.swift
import Foundation
import CryptoKit
import CommonCrypto

/**
 * Secure Communication Utility for iOS Master App
 * Matches the encryption implementation in Porsche E-Claims React Native app
 */

public class SecureCommunication {
    
    // MARK: - Configuration
    
    private let sharedSecret: String
    private let iterations: Int = 1000
    
    public init(sharedSecret: String = "P0rsch3-ECl41ms-S3cur3-K3y-D3v3l0pm3nt-2026!@#") {
        self.sharedSecret = sharedSecret
    }
    
    // MARK: - Public Methods
    
    /**
     * Encrypt vehicle data for secure deeplink transmission
     */
    public func encryptVehicleData(_ vehicleData: [String: Any]) throws -> String {
        let payload = [
            "vehicleData": vehicleData,
            "source": "master-app",
            "version": "1.0"
        ]
        
        return try encryptData(payload)
    }
    
    /**
     * Create secure authentication token
     */
    public func createSecureAuthToken(userId: String, expirationMinutes: Int = 60) throws -> String {
        let authData: [String: Any] = [
            "userId": userId,
            "expiresAt": Date().timeIntervalSince1970 * 1000 + Double(expirationMinutes * 60 * 1000),
            "issuedAt": Date().timeIntervalSince1970 * 1000,
            "nonce": UUID().uuidString,
            "iss": "porsche-master-app"
        ]
        
        return try encryptData(authData)
    }
    
    /**
     * Generate secure deeplink URL
     */
    public func generateSecureDeeplink(vehicleData: [String: Any], authToken: String? = nil) throws -> URL {
        let encryptedVehicleData = try encryptVehicleData(vehicleData)
        
        var components = URLComponents()
        components.scheme = "porscheeclaims"
        components.host = "vehicles"
        
        var queryItems = [
            URLQueryItem(name: "vehicleData", value: encryptedVehicleData)
        ]
        
        if let authToken = authToken {
            queryItems.append(URLQueryItem(name: "token", value: authToken))
        }
        
        components.queryItems = queryItems
        
        guard let url = components.url else {
            throw SecureCommunicationError.urlGenerationFailed
        }
        
        return url
    }
    
    // MARK: - Private Methods
    
    /**
     * Encrypt data using AES-256-CBC with PBKDF2 key derivation
     */
    private func encryptData(_ data: [String: Any]) throws -> String {
        let jsonData = try JSONSerialization.data(withJSONObject: data, options: [])
        let timestamp = Int64(Date().timeIntervalSince1970 * 1000)
        
        // Generate random IV
        let iv = Data((0..<16).map { _ in UInt8.random(in: 0...255) })
        
        // Derive key using PBKDF2
        let key = try deriveKey(password: sharedSecret, salt: iv)
        
        // Encrypt data
        let encryptedData = try aesEncrypt(data: jsonData, key: key, iv: iv)
        
        // Create signature
        let signature = try createSignature(data: encryptedData.base64EncodedString(), timestamp: timestamp)
        
        // Create payload
        let payload: [String: Any] = [
            "data": encryptedData.base64EncodedString(),
            "iv": iv.base64EncodedString(),
            "timestamp": timestamp,
            "signature": signature
        ]
        
        let payloadData = try JSONSerialization.data(withJSONObject: payload, options: [])
        return payloadData.base64EncodedString()
    }
    
    /**
     * Derive encryption key using PBKDF2
     */
    private func deriveKey(password: String, salt: Data) throws -> Data {
        guard let passwordData = password.data(using: .utf8) else {
            throw SecureCommunicationError.invalidPassword
        }
        
        var derivedKey = Data(count: 32) // 256 bits
        let result = derivedKey.withUnsafeMutableBytes { derivedKeyBytes in
            salt.withUnsafeBytes { saltBytes in
                CCKeyDerivationPBKDF(
                    CCPBKDFAlgorithm(kCCPBKDF2),
                    passwordData.withUnsafeBytes { $0.baseAddress?.assumingMemoryBound(to: Int8.self) },
                    passwordData.count,
                    saltBytes.baseAddress?.assumingMemoryBound(to: UInt8.self),
                    salt.count,
                    CCPseudoRandomAlgorithm(kCCPRFHmacAlgSHA256),
                    UInt32(iterations),
                    derivedKeyBytes.baseAddress?.assumingMemoryBound(to: UInt8.self),
                    derivedKey.count
                )
            }
        }
        
        guard result == kCCSuccess else {
            throw SecureCommunicationError.keyDerivationFailed
        }
        
        return derivedKey
    }
    
    /**
     * AES-256-CBC encryption
     */
    private func aesEncrypt(data: Data, key: Data, iv: Data) throws -> Data {
        var encryptedData = Data(count: data.count + kCCBlockSizeAES128)
        var encryptedDataLength: size_t = 0
        
        let status = encryptedData.withUnsafeMutableBytes { encryptedBytes in
            data.withUnsafeBytes { dataBytes in
                key.withUnsafeBytes { keyBytes in
                    iv.withUnsafeBytes { ivBytes in
                        CCCrypt(
                            CCOperation(kCCEncrypt),
                            CCAlgorithm(kCCAlgorithmAES),
                            CCOptions(kCCOptionPKCS7Padding),
                            keyBytes.baseAddress,
                            key.count,
                            ivBytes.baseAddress,
                            dataBytes.baseAddress,
                            data.count,
                            encryptedBytes.baseAddress,
                            encryptedData.count,
                            &encryptedDataLength
                        )
                    }
                }
            }
        }
        
        guard status == kCCSuccess else {
            throw SecureCommunicationError.encryptionFailed
        }
        
        return encryptedData.prefix(encryptedDataLength)
    }
    
    /**
     * Create HMAC-SHA256 signature
     */
    private func createSignature(data: String, timestamp: Int64) throws -> String {
        let message = "\(data):\(timestamp)"
        guard let messageData = message.data(using: .utf8),
              let keyData = sharedSecret.data(using: .utf8) else {
            throw SecureCommunicationError.signatureCreationFailed
        }
        
        let signature = HMAC<SHA256>.authenticationCode(for: messageData, using: SymmetricKey(data: keyData))
        return Data(signature).map { String(format: "%02hhx", $0) }.joined()
    }
}

// MARK: - Convenience Extensions

public extension SecureCommunication {
    
    /**
     * Create deeplink for Porsche vehicle
     */
    func createPorscheVehicleDeeplink(
        make: String = "Porsche",
        model: String,
        year: Int,
        vin: String,
        licensePlate: String? = nil,
        color: String? = nil,
        insuranceCompany: String? = nil,
        policyNumber: String? = nil,
        userId: String? = nil
    ) throws -> URL {
        
        var vehicleData: [String: Any] = [
            "make": make,
            "model": model,
            "year": year,
            "vin": vin
        ]
        
        if let licensePlate = licensePlate { vehicleData["licensePlate"] = licensePlate }
        if let color = color { vehicleData["color"] = color }
        if let insuranceCompany = insuranceCompany { vehicleData["insuranceCompany"] = insuranceCompany }
        if let policyNumber = policyNumber { vehicleData["policyNumber"] = policyNumber }
        
        let authToken = try userId.map { try createSecureAuthToken(userId: $0) }
        
        return try generateSecureDeeplink(vehicleData: vehicleData, authToken: authToken)
    }
}

// MARK: - Error Types

public enum SecureCommunicationError: Error, LocalizedError {
    case invalidPassword
    case keyDerivationFailed
    case encryptionFailed
    case signatureCreationFailed
    case urlGenerationFailed
    
    public var errorDescription: String? {
        switch self {
        case .invalidPassword:
            return "Invalid password for encryption"
        case .keyDerivationFailed:
            return "Failed to derive encryption key"
        case .encryptionFailed:
            return "Failed to encrypt data"
        case .signatureCreationFailed:
            return "Failed to create signature"
        case .urlGenerationFailed:
            return "Failed to generate URL"
        }
    }
}

// MARK: - Usage Example

/*
// Example usage in iOS Master App:

let secureCom = SecureCommunication()

do {
    // Create secure deeplink for Porsche vehicle
    let deeplinkURL = try secureCom.createPorscheVehicleDeeplink(
        model: "911 Carrera",
        year: 2024,
        vin: "WP0CA2A89KS123456",
        licensePlate: "P-EC 123",
        color: "Guards Red",
        userId: "user123"
    )
    
    // Open E-Claims app with secure deeplink
    if UIApplication.shared.canOpenURL(deeplinkURL) {
        UIApplication.shared.open(deeplinkURL, options: [:], completionHandler: nil)
    }
    
} catch {
    print("Failed to create secure deeplink: \(error.localizedDescription)")
}

// Manual vehicle data creation:
let vehicleData = [
    "make": "Porsche",
    "model": "Taycan",
    "year": 2024,
    "vin": "WP0ZZZ9999S123456"
] as [String : Any]

do {
    let authToken = try secureCom.createSecureAuthToken(userId: "demo-user", expirationMinutes: 30)
    let secureURL = try secureCom.generateSecureDeeplink(vehicleData: vehicleData, authToken: authToken)
    
    UIApplication.shared.open(secureURL)
} catch {
    print("Error: \(error)")
}
*/

File: platform-integration/android/SecureCommunication.kt
package com.porsche.eclaims.security

import android.util.Base64
import org.json.JSONObject
import java.nio.charset.StandardCharsets
import java.security.SecureRandom
import javax.crypto.Cipher
import javax.crypto.Mac
import javax.crypto.SecretKeyFactory
import javax.crypto.spec.IvParameterSpec
import javax.crypto.spec.PBEKeySpec
import javax.crypto.spec.SecretKeySpec
import java.util.*

/**
 * JSONObject extension to convert to Map
 */
fun JSONObject.toMap(): Map<String, Any> {
    val map = mutableMapOf<String, Any>()
    val keys = keys()
    while (keys.hasNext()) {
        val key = keys.next()
        val value = get(key)
        map[key] = when (value) {
            is JSONObject -> value.toMap()
            else -> value
        }
    }
    return map
}

/**
 * Secure Communication Utility for Android Master App
 * Matches the encryption implementation in Porsche E-Claims React Native app
 *
 * NOTE: This file is a template. Use scripts/generate-platform-code.sh to generate
 * a version with your actual shared secret injected.
 */
class SecureCommunication @JvmOverloads constructor(
    private val sharedSecret: String = "P0rsch3-ECl41ms-S3cur3-K3y-D3v3l0pm3nt-2026!@#"
) {
    
    companion object {
        private const val AES_TRANSFORMATION = "AES/CBC/PKCS5Padding"
        private const val HMAC_ALGORITHM = "HmacSHA256"
        private const val PBKDF2_ALGORITHM = "PBKDF2WithHmacSHA256"
        private const val KEY_LENGTH = 256
        private const val ITERATIONS = 1000
        private const val IV_LENGTH = 16
    }
    
    /**
     * Encrypt vehicle data for secure deeplink transmission
     */
    fun encryptVehicleData(vehicleData: Map<String, Any>): String {
        val payload = mapOf(
            "vehicleData" to vehicleData,
            "source" to "master-app",
            "version" to "1.0"
        )
        
        return encryptData(payload)
    }
    
    /**
     * Create secure authentication token
     */
    fun createSecureAuthToken(userId: String, expirationMinutes: Int = 60): String {
        val authData = mapOf(
            "userId" to userId,
            "expiresAt" to (System.currentTimeMillis() + expirationMinutes * 60 * 1000),
            "issuedAt" to System.currentTimeMillis(),
            "nonce" to UUID.randomUUID().toString(),
            "iss" to "porsche-master-app"
        )
        
        return encryptData(authData)
    }
    
    /**
     * Generate secure deeplink URL
     */
    fun generateSecureDeeplink(vehicleData: Map<String, Any>, authToken: String? = null): String {
        val encryptedVehicleData = encryptVehicleData(vehicleData)
        
        val urlBuilder = StringBuilder("porscheeclaims://vehicles?vehicleData=")
        urlBuilder.append(android.net.Uri.encode(encryptedVehicleData))
        
        authToken?.let {
            urlBuilder.append("&token=").append(android.net.Uri.encode(it))
        }
        
        return urlBuilder.toString()
    }
    
    /**
     * Generate legacy deeplink URL (for backward compatibility)
     */
    fun generateLegacyDeeplink(vehicleData: Map<String, Any>, authToken: String? = null): String {
        val legacyVehicleData = Base64.encodeToString(
            JSONObject(vehicleData).toString().toByteArray(StandardCharsets.UTF_8), 
            Base64.NO_WRAP
        )
        
        val urlBuilder = StringBuilder("porscheeclaims://vehicles?vehicleData=")
        urlBuilder.append(android.net.Uri.encode(legacyVehicleData))
        
        authToken?.let {
            urlBuilder.append("&token=").append(android.net.Uri.encode(it))
        }
        
        return urlBuilder.toString()
    }
    
    /**
     * Check if data appears to be encrypted (vs legacy base64)
     */
    fun isEncryptedData(data: String): Boolean {
        return try {
            val decoded = JSONObject(String(Base64.decode(data, Base64.NO_WRAP), StandardCharsets.UTF_8))
            decoded.has("data") && decoded.has("iv") && decoded.has("timestamp") && decoded.has("signature")
        } catch (e: Exception) {
            false
        }
    }
    
    /**
     * Smart vehicle data extraction with automatic format detection
     * Use this method to handle both encrypted and legacy formats
     */
    fun smartExtractVehicleData(data: String): Map<String, Any>? {
        return if (isEncryptedData(data)) {
            // Handle encrypted data
            try {
                val decrypted = decryptData(data)
                @Suppress("UNCHECKED_CAST")
                decrypted["vehicleData"] as? Map<String, Any>
            } catch (e: Exception) {
                android.util.Log.w("SecureCommunication", "Encrypted data extraction failed", e)
                null
            }
        } else {
            // Handle legacy base64 data
            try {
                val jsonString = String(Base64.decode(data, Base64.NO_WRAP), StandardCharsets.UTF_8)
                val jsonObject = JSONObject(jsonString)
                jsonObject.toMap()
            } catch (e: Exception) {
                android.util.Log.e("SecureCommunication", "Legacy base64 decoding failed", e)
                null
            }
        }
    }
    
    /**
     * Decrypt data (for encrypted format)
     */
    fun decryptData(encryptedData: String): Map<String, Any> {
        val payloadJson = String(Base64.decode(encryptedData, Base64.NO_WRAP), StandardCharsets.UTF_8)
        val payload = JSONObject(payloadJson)
        
        val data = payload.getString("data")
        val iv = Base64.decode(payload.getString("iv"), Base64.NO_WRAP)
        val timestamp = payload.getLong("timestamp")
        val signature = payload.getString("signature")
        
        // Validate timestamp (5 minutes max age)
        val maxAge = 5 * 60 * 1000L
        if (System.currentTimeMillis() - timestamp > maxAge) {
            throw SecurityException("Data expired")
        }
        
        // Verify signature
        val expectedSignature = createSignature(data, timestamp)
        if (signature != expectedSignature) {
            throw SecurityException("Data integrity check failed")
        }
        
        // Decrypt
        val key = deriveKey(sharedSecret, iv)
        val decryptedBytes = aesDecrypt(Base64.decode(data, Base64.NO_WRAP), key, iv)
        val decryptedJson = String(decryptedBytes, StandardCharsets.UTF_8)
        
        return JSONObject(decryptedJson).toMap()
    }
    
    /**
     * Encrypt data using AES-256-CBC with PBKDF2 key derivation
     */
    private fun encryptData(data: Map<String, Any>): String {
        val jsonString = JSONObject(data).toString()
        val timestamp = System.currentTimeMillis()
        
        // Generate random IV
        val iv = ByteArray(IV_LENGTH)
        SecureRandom().nextBytes(iv)
        
        // Derive key using PBKDF2
        val key = deriveKey(sharedSecret, iv)
        
        // Encrypt data
        val encryptedData = aesEncrypt(jsonString.toByteArray(StandardCharsets.UTF_8), key, iv)
        
        // Create signature
        val encryptedBase64 = Base64.encodeToString(encryptedData, Base64.NO_WRAP)
        val signature = createSignature(encryptedBase64, timestamp)
        
        // Create payload
        val payload = mapOf(
            "data" to encryptedBase64,
            "iv" to Base64.encodeToString(iv, Base64.NO_WRAP),
            "timestamp" to timestamp,
            "signature" to signature
        )
        
        val payloadJson = JSONObject(payload).toString()
        return Base64.encodeToString(payloadJson.toByteArray(StandardCharsets.UTF_8), Base64.NO_WRAP)
    }
    
    /**
     * Derive encryption key using PBKDF2
     */
    private fun deriveKey(password: String, salt: ByteArray): ByteArray {
        val spec = PBEKeySpec(password.toCharArray(), salt, ITERATIONS, KEY_LENGTH)
        val factory = SecretKeyFactory.getInstance(PBKDF2_ALGORITHM)
        return factory.generateSecret(spec).encoded
    }
    
    /**
     * AES-256-CBC encryption
     */
    private fun aesEncrypt(data: ByteArray, key: ByteArray, iv: ByteArray): ByteArray {
        val cipher = Cipher.getInstance(AES_TRANSFORMATION)
        val secretKey = SecretKeySpec(key, "AES")
        val ivSpec = IvParameterSpec(iv)
        
        cipher.init(Cipher.ENCRYPT_MODE, secretKey, ivSpec)
        return cipher.doFinal(data)
    }
    
    /**
     * AES-256-CBC decryption
     */
    private fun aesDecrypt(encryptedData: ByteArray, key: ByteArray, iv: ByteArray): ByteArray {
        val cipher = Cipher.getInstance(AES_TRANSFORMATION)
        val secretKey = SecretKeySpec(key, "AES")
        val ivSpec = IvParameterSpec(iv)
        
        cipher.init(Cipher.DECRYPT_MODE, secretKey, ivSpec)
        return cipher.doFinal(encryptedData)
    }
    
    /**
     * Create HMAC-SHA256 signature
     */
    private fun createSignature(data: String, timestamp: Long): String {
        val message = "$data:$timestamp"
        val mac = Mac.getInstance(HMAC_ALGORITHM)
        val secretKey = SecretKeySpec(sharedSecret.toByteArray(StandardCharsets.UTF_8), HMAC_ALGORITHM)
        
        mac.init(secretKey)
        val signature = mac.doFinal(message.toByteArray(StandardCharsets.UTF_8))
        
        return signature.joinToString("") { "%02x".format(it) }
    }
}

/**
 * Convenience extension functions
 */
object SecureCommunicationExtensions {
    
    /**
     * Create secure deeplink for Porsche vehicle
     */
    fun SecureCommunication.createPorscheVehicleDeeplink(
        make: String = "Porsche",
        model: String,
        year: Int,
        vin: String,
        licensePlate: String? = null,
        color: String? = null,
        insuranceCompany: String? = null,
        policyNumber: String? = null,
        userId: String? = null,
        useEncryption: Boolean = true
    ): String {
        
        val vehicleData = mutableMapOf<String, Any>(
            "make" to make,
            "model" to model,
            "year" to year,
            "vin" to vin
        )
        
        licensePlate?.let { vehicleData["licensePlate"] = it }
        color?.let { vehicleData["color"] = it }
        insuranceCompany?.let { vehicleData["insuranceCompany"] = it }
        policyNumber?.let { vehicleData["policyNumber"] = it }
        
        val authToken = userId?.let { 
            if (useEncryption) createSecureAuthToken(it) else createLegacyAuthToken(it)
        }
        
        return if (useEncryption) {
            generateSecureDeeplink(vehicleData, authToken)
        } else {
            generateLegacyDeeplink(vehicleData, authToken)
        }
    }
    
    /**
     * Create legacy deeplink for Porsche vehicle (backward compatibility)
     */
    fun SecureCommunication.createPorscheLegacyDeeplink(
        make: String = "Porsche",
        model: String,
        year: Int,
        vin: String,
        licensePlate: String? = null,
        color: String? = null,
        insuranceCompany: String? = null,
        policyNumber: String? = null,
        userId: String? = null
    ): String {
        return createPorscheVehicleDeeplink(
            make, model, year, vin, licensePlate, color, 
            insuranceCompany, policyNumber, userId, useEncryption = false
        )
    }
    
    /**
     * Create legacy auth token (base64 JSON)
     */
    private fun SecureCommunication.createLegacyAuthToken(userId: String, expirationHours: Int = 24): String {
        val authData = mapOf(
            "userId" to userId,
            "expiresAt" to (System.currentTimeMillis() + expirationHours * 60 * 60 * 1000),
            "sessionId" to UUID.randomUUID().toString()
        )
        
        val jsonString = JSONObject(authData).toString()
        return Base64.encodeToString(jsonString.toByteArray(StandardCharsets.UTF_8), Base64.NO_WRAP)
    }
}

// Data classes for structured vehicle data
data class PorscheVehicle(
    val make: String = "Porsche",
    val model: String,
    val year: Int,
    val vin: String,
    val licensePlate: String? = null,
    val color: String? = null,
    val insuranceCompany: String? = null,
    val policyNumber: String? = null
) {
    fun toMap(): Map<String, Any> {
        val map = mutableMapOf<String, Any>(
            "make" to make,
            "model" to model,
            "year" to year,
            "vin" to vin
        )
        
        licensePlate?.let { map["licensePlate"] = it }
        color?.let { map["color"] = it }
        insuranceCompany?.let { map["insuranceCompany"] = it }
        policyNumber?.let { map["policyNumber"] = it }
        
        return map
    }
}

/*
Usage Examples:

// RECOMMENDED: Use secure encryption (default)
val secureCom = SecureCommunication()

val vehicle = PorscheVehicle(
    model = "911 Carrera",
    year = 2024,
    vin = "WP0CA2A89KS123456",
    licensePlate = "P-EC 123",
    color = "Guards Red"
)

try {
    // Create secure encrypted deeplink
    val secureDeeplinkUrl = SecureCommunicationExtensions.run {
        secureCom.createPorscheVehicleDeeplink(
            model = vehicle.model,
            year = vehicle.year,
            vin = vehicle.vin,
            licensePlate = vehicle.licensePlate,
            color = vehicle.color,
            userId = "user123",
            useEncryption = true  // Uses strong encryption
        )
    }
    
    // Launch E-Claims app with secure deeplink
    val intent = Intent(Intent.ACTION_VIEW, Uri.parse(secureDeeplinkUrl))
    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
    
    if (intent.resolveActivity(context.packageManager) != null) {
        context.startActivity(intent)
    } else {
        // Handle case where E-Claims app is not installed
        Log.e("SecureCommunication", "E-Claims app not found")
    }
    
} catch (e: Exception) {
    Log.e("SecureCommunication", "Failed to create secure deeplink", e)
}

// LEGACY: For backward compatibility only
try {
    val legacyDeeplinkUrl = SecureCommunicationExtensions.run {
        secureCom.createPorscheLegacyDeeplink(
            model = vehicle.model,
            year = vehicle.year,
            vin = vehicle.vin,
            licensePlate = vehicle.licensePlate,
            color = vehicle.color,
            userId = "user123"
        )
    }
    
    val intent = Intent(Intent.ACTION_VIEW, Uri.parse(legacyDeeplinkUrl))
    context.startActivity(intent)
    
} catch (e: Exception) {
    Log.e("SecureCommunication", "Failed to create legacy deeplink", e)
}

// SMART EXTRACTION: Handle received data (if building a receiver app)
val receivedData = "..." // Data from deeplink
val extractedVehicle = secureCom.smartExtractVehicleData(receivedData)
if (extractedVehicle != null) {
    Log.d("SecureCommunication", "Vehicle received: ${extractedVehicle["make"]} ${extractedVehicle["model"]}")
} else {
    Log.e("SecureCommunication", "Failed to extract vehicle data")
}

// FORMAT DETECTION
val isEncrypted = secureCom.isEncryptedData(receivedData)
Log.d("SecureCommunication", "Data format: ${if (isEncrypted) "Encrypted" else "Legacy Base64"}")

// In Activity or Fragment:
class MainActivity : AppCompatActivity() {
    
    private val secureCom = SecureCommunication()
    
    fun launchEClaimsWithVehicle(vehicle: PorscheVehicle, userId: String, useSecureFormat: Boolean = true) {
        try {
            val deeplinkUrl = SecureCommunicationExtensions.run {
                secureCom.createPorscheVehicleDeeplink(
                    model = vehicle.model,
                    year = vehicle.year,
                    vin = vehicle.vin,
                    licensePlate = vehicle.licensePlate,
                    color = vehicle.color,
                    userId = userId,
                    useEncryption = useSecureFormat
                )
            }
            
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(deeplinkUrl))
            intent.setPackage("com.deactech.porscheeclaims") // Specific package
            
            if (intent.resolveActivity(packageManager) != null) {
                startActivity(intent)
            } else {
                // Show dialog to install E-Claims app
                showInstallEClaimsDialog()
            }
            
        } catch (e: Exception) {
            Log.e("MainActivity", "Failed to launch E-Claims app", e)
            // Show error to user
        }
    }
    
    private fun showInstallEClaimsDialog() {
        AlertDialog.Builder(this)
            .setTitle("Install E-Claims App")
            .setMessage("The Porsche E-Claims app is required. Would you like to install it?")
            .setPositiveButton("Install") { _, _ ->
                // Open Play Store
                val playStoreIntent = Intent(Intent.ACTION_VIEW).apply {
                    data = Uri.parse("market://details?id=com.deactech.porscheeclaims")
                }
                startActivity(playStoreIntent)
            }
            .setNegativeButton("Cancel", null)
            .show()
    }
}
*/